keyring/backends/Windows.py

```diff
@@ -1,46 +1,254 @@
-import functools
+import json
 import logging
+import sys
+from weakref import WeakKeyDictionary
 
-from ..util import properties
 from ..backend import KeyringBackend
 from ..credentials import SimpleCredential
-from ..errors import PasswordDeleteError, ExceptionRaisedContext
+from ..errors import ExceptionRaisedContext, PasswordDeleteError
+from ..util import properties
+
+MAX_PASSWORD_BYTES = 2 ** 20
+TARGET_SHARD = "{target}-shard-{n:04}"
+ATTRIBUTE_KEYWORD = "keyring_metadata"
 
 
+# region standalone replacement for win32cred
 with ExceptionRaisedContext() as missing_deps:
+    import cffi
+
+    if sys.platform != "win32":
+        raise EnvironmentError(f"Windows backend requires sys.platform = 'win32' but {sys.platform=}")
+    else:
+        ffi = cffi.FFI()
```

Comment:
I would suggest moving all of the CFFI stuff into an `api` module, similar to what the `macOS` backend does.

Reply:
I will study that backend.  I definitely want it in a separate module.  I wasn't sure what the right way was to do that.

Reply:
What you did was pretty good. It's a little awkward that the `macOS` capitalization made it to `windowsOS` (which isn't a thing), but that's easily corrected. The main thing I would do differently is to keep everything for Windows in a single package, like so:

```
$ git mv keyring/backends/Windows.py keyring/backends/Windows/__init__.py
$ git mv keyring/backends/windowsOS/api.py keyring/backends/Windows
$ git rm keyring/backends/windowsOS/__init__.py
```

And then change the import to use `from . import api`.